lxd/instance/drivers/driver_qemu.go

```diff
@@ -1600,6 +1600,105 @@ func (d *qemu) start(stateful bool, op *operationlock.InstanceOperation) error {
 	return nil
 }
 
+func (d *qemu) setupSEV(sevOpts *qemuSevOpts, fdFiles *[]*os.File) error {
+	if d.architecture != osarch.ARCH_64BIT_INTEL_X86 {
+		return errors.New("AMD SEV support is only available on x86_64 systems")
+	}
+
+	qemuPath, _, err := d.qemuArchConfig(d.architecture)
+	if err != nil {
+		return err
+	}
+
+	// Get the QEMU features to check if AMD SEV is supported.
+	features, err := d.checkFeatures(d.architecture, qemuPath)
+	if err != nil {
+		return err
+	}
+
+	_, smeFound := features["sme"]
+	sev, sevFound := features["sev"]
+	if !smeFound || !sevFound {
+		return errors.New("AMD SEV is not supported by the host")
+	}
+
+	// Get the SEV guest `cbitpos` and `reducedPhysBits`.
+	sevCapabilities, ok := sev.(qmp.AMDSEVCapabilities)
+	if !ok {
+		return errors.New("Failed to get the guest `sev` capabilities")
+	}
+
+	cbitpos := sevCapabilities.CBitPos
+	reducedPhysBits := sevCapabilities.ReducedPhysBits
+
+	// Write user's dh-cert and session-data to file descriptors.
+	var dhCertFD, sessionDataFD int
+	var dhCertFDOk, sessionDataFDOk bool
```

Comment:
Do we need these vars, if the associated FD variables are >0 then they are OK I believe?